src/plugins/gs-plugin-eos.c

```diff
@@ -1019,3 +1019,43 @@ gs_plugin_add_updates (GsPlugin *plugin,
 	 * this will still work correctly */
 	return add_updates (plugin, list, cancellable, error);
 }
+
+static gboolean
+filter_apps_by_prefix (GsApp *app, const char **prefixes)
+{
+	gboolean keep = FALSE;
+	guint i;
+	for (i = 0; prefixes[i] != NULL; ++i) {
+		if (g_str_has_prefix (gs_app_get_id (app), prefixes[i]))
+			return TRUE;
+	}
+	return FALSE;
+}
+
+static gboolean
+ensure_only_fnde_popular_apps (GsPlugin *plugin,
+			       GsAppList *list,
+			       GCancellable *cancellable,
+			       GError **error)
+{
+	guint i;
+	GsApp *app;
+	const char *prefixes[] = {"com.endlessm.fnde_", "com.endlessm.teachers_",
```

Comment:
I think it would be better to be more specific and use "com.endlessm.teachers_portal_".